libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -63,27 +63,21 @@
 
 extern const AP_HAL::HAL& hal;
 
-// update the _rp_state of the sensor
 void AP_Proximity_RPLidarA2::update(void)
 {
     if (_uart == nullptr) {
         return;
     }
 
-    // initialise sensor if necessary
-    if (!_initialised) {
-        _initialised = initialise();    //returns true if everything initialized properly
-    }
-
-    // if LIDAR in known state
-    if (_initialised) {
-        get_readings();
-    }
+    get_readings();
 
     // check for timeout and set health status
-    if ((_last_distance_received_ms == 0) || (AP_HAL::millis() - _last_distance_received_ms > COMM_ACTIVITY_TIMEOUT_MS)) {
+    if (AP_HAL::millis() - _last_distance_received_ms > COMM_ACTIVITY_TIMEOUT_MS) {
         set_status(AP_Proximity::Status::NoData);
         Debug(1, "LIDAR NO DATA");
+        if (AP_HAL::millis() - _last_reset_ms > 1000) {
```

Comment:
can't you juste const auto now = AP_HAL::millis() ?

Reply:
Done, thanks!